lib/spack/llnl/util/tty/log.py

```diff
@@ -449,7 +449,7 @@ def _writer_daemon(self, stdin):
                     # Without the last parameter (timeout) select will
```

Comment:
This comment explicitly states, that the last parameter may not be safely removed, so don't do that (or at the very least remove the comment). Does setting an explicit timeout >0 also fix your problem?

Reply:
I think a large timeout can indeed hide the problem. However, I seriously question if the comment above is still true.